### PR TITLE
9063 make account

### LINKED ISSUE #9063: orchestration - chain - provide well-known chain
## What is the Problem Being Solved?
Allow an object derived from a `ZcfSeat` to return a `Chain` object based on a `chainName` which exposes common chain actions, including:
- Creating or retrieving an ocap to control an ICA account hosted on that chain
- Making an ICQ query to that chain
- Any custom chain actions

For chains _well-known_ to the API, the user should not have to specify IBC connection info.
```[tasklist]
### Well Known Chains
- [x] orchestration - create chain - Osmosis
- [x] Umee (UX Chain)
- [x] DYDX
- [x] Stride
- [x] Secret
- [x] Neutron
- [x] Noble
- [x] Celestia
- [x] Cosmos Hub
- [x] Omniflix
- [x] Stargaze
```

Additionally, allow for chains unknown to the API additional details like IBC connection info may be needed.

```[tasklist]
### Dependencies
- [ ] https://github.com/Agoric/agoric-sdk/issues/9187
```

### PR DESCRIPTION
**Pull request by turadg**
closes: #9063

## Description

Wrapping up lose ends to close #9063. Whatever's left belongs in a new ticket.

### CODE REVIEW

File: packages/orchestration/src/facade.js [hunk 1]
```diff
@@ -102,11 +102,9 @@ const makeRemoteChainFacade = (
   chainInfo,
   { orchestration, timer, zcf, zone },
 ) => {
-  const name = chainInfo.chainId;
```
Comment (dckc) [inline packages/orchestration/src/facade.js:105]: yeah; that was weird

File: packages/orchestration/src/chain-info.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ import fetchedChainInfo from './fetched-chain-info.js';
 const knownChains = /** @satisfies {Record<string, ChainInfo>} */ (
   harden({
     ...fetchedChainInfo,
-    agoric: {
+    // XXX does not have useful connections
+    // UNTIL https://github.com/Agoric/agoric-sdk/issues/9492
+    agoriclocal: {
```
Comment (dckc) [inline packages/orchestration/src/chain-info.js:15]: using `agoriclocal` as both a chain id and a chain name made me scratch my head a bit. `aglocal`? not sure that helps much. The XXX suffices, I suppose.

File: packages/orchestration/test/chain-info.test.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ test('chain-info', async t => {
   });
   t.deepEqual(chainNames.keys(), [
     'agoric',
+    'agoriclocal',
```
Comment (dckc) [inline packages/orchestration/test/chain-info.test.js:18]: having something that's also a chainid show up here is a bit of a head-tilter. XXX again?

File: packages/orchestration/test/examples/sendAnywhere.test.ts [hunk 4]
```diff
@@ -164,7 +164,7 @@ test('send using arbitrary chain info', async t => {
       '@type': '/ibc.applications.transfer.v1.MsgTransfer',
       receiver: 'cosmos1destAddr',
       sender: execAddr,
-      sourceChannel: 'channel-1',
+      sourceChannel: 'channel-5',
```
Comment (dckc) [inline packages/orchestration/test/examples/sendAnywhere.test.ts:167]: :clap: I realized that should be higher the other day but didn't do anything about it.

File: packages/orchestration/src/examples/swapExample.contract.js [hunk 5]
```diff
@@ -105,15 +105,15 @@ export const start = async (zcf, privateArgs, baggage) => {
     async (/** @type {Orchestrator} */ orch, { zcf }, seat, offerArgs) => {
       const { give } = seat.getProposal();
 
-      const celestia = await orch.getChain('celestia');
+      const omni = await orch.getChain('omniflixhub');
```
Comment (dckc) [inline packages/orchestration/src/examples/swapExample.contract.js:108]: @dtribble I think you presented this celestia example publicly and the choice of chain was not arbitrary, so this is in some sense product API surface. But we're using more real data now, and there is no connection to celestia yet. Be advised: we're changing this example to one of [Agoric's current peer zones](https://mapofzones.com/zones/agoric-3/peers?columnKey=ibcVolumeIn&period=24h).

cc @mitdralla

File: packages/orchestration/src/facade.js [hunk 6]
```diff
@@ -113,14 +116,12 @@ const makeRemoteChainFacade = (
...
     makeAccount: async () => {
-      // FIXME look up real values
-      // UNTIL https://github.com/Agoric/agoric-sdk/issues/9063
-      const hostConnectionId = 'connection-1';
-      const controllerConnectionId = 'connection-2';
-
       const icaAccount = await E(orchestration).makeAccount(
-        hostConnectionId,
-        controllerConnectionId,
+        // XXX IBCConnectionInfo concessions for JSON encoding
```
Comment (dckc) [inline packages/orchestration/src/facade.js:121]: I don't understand that XXX comment; elaborate a bit, please?

File: packages/orchestration/test/examples/stake-atom.contract.test.ts [hunk 7]
```diff
@@ -56,7 +57,7 @@ test('makeAccount, deposit, withdraw', async t => {
   const address = await E(account).getAddress();
   // XXX address.address is weird
   //   t.regex(address.address, /agoric1/);
-  t.like(address, { chainId: 'FIXME', addressEncoding: 'bech32' });
+  t.like(address, { chainId: 'cosmoshub-4', addressEncoding: 'bech32' });
```
Comment (dckc) [inline packages/orchestration/test/examples/stake-atom.contract.test.ts:60]: :clap: nice to see these getting fleshed out

File: packages/boot/test/bootstrapTests/vat-orchestration.test.ts [hunk 8]
```diff
@@ -95,7 +96,7 @@ test('makeAccount returns an ICA connection', async t => {
   t.regex(remoteAddress, /icahost/);
   t.regex(localAddress, /icacontroller/);
   t.regex(chainAddress.address, /cosmos1/);
-  t.regex(chainAddress.chainId, /FIXME/); // TODO, use a real chainId #9063
+  t.is(chainAddress.chainId, 'somechain-1');
```
Comment (dckc) [inline packages/boot/test/bootstrapTests/vat-orchestration.test.ts:99]: `somechain-1` isn't a real chainId in the sense of the fetched chain registry data. I guess that's OK?
This test clearly doesn't actually require a real chainId.

**Review by dckc — APPROVED** [review]
very nice to see the whole design coming together

File: packages/orchestration/src/facade.js (hunk 6 above)
Comment (turadg) [reply packages/orchestration/src/facade.js:121]: We had a string template type for these kinds of keys but when I switched to the fetcher that spits out JSON, I had to widen them to strings.

I hope to come back to that before release

File: packages/orchestration/src/fetched-chain-info.js [hunk 9]
```diff
@@ -1,5 +1,141 @@
 /** @file Generated by fetch-chain-info.ts */
 export default {
+  agoric: {
+    chainId: 'agoric-3',
+    stakingTokens: [
+      {
+        denom: 'ubld',
+      },
+    ],
```
Comment (0xpatrickdev) [inline packages/orchestration/src/fetched-chain-info.js:9]: Tangential to the PR, but i think this should be `bondDenom: string` instead of an arary.

**Review by 0xpatrickdev — APPROVED** [review]

**Pull request merged**